cimbar/cimbar.py

```diff
@@ -114,41 +119,76 @@ def _get_decoder_stream(outfile, ecc, fountain):
     return reed_solomon_stream(f, ecc, mode='write', on_failure=on_rss_failure) if ecc else f
 
 
-def decode_iter(src_image, dark, force_preprocess, deskew, auto_dewarp):
-    should_preprocess = force_preprocess
+def compute_tint(img, dark):
+    def update(c, r, g, b):
+        c['r'] = max(c['r'], r)
+        c['g'] = max(c['g'], g)
+        c['b'] = max(c['b'], b)
+
+    cc = {}
+    cc['r'] = cc['g'] = cc['b'] = 1
+
+    if dark:
+        pos = [(28, 28), (28, 992), (992, 28)]
+    else:
+        pos = [(67, 0), (0, 67), (945, 0), (0, 945)]
+
+    for x, y in pos:
+        iblock = img.crop((x, y, x + 4, y + 4))
+        r, g, b = avg_color(iblock)
+        update(cc, *avg_color(iblock))
+
+    print(f'tint is {cc}')
+    return cc['r'], cc['g'], cc['b']
+
+
+def _decode_iter(ct, img, color_img):
```

Comment:
This is split out because one of the potential tricks is to run the decode twice -- computing the color distribution (or other statistics) and using those to inform the correction matrix for the 2nd pass.

Ideally the CCM would work across frames. If so, it could be computed once every N frames and used for real-time decoding.